code/mob/living/critter/small_animal.dm

```diff
@@ -677,6 +689,43 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 					src.delStatus("weakened")
 					src.icon_state = src.dogtype
 
+	Life(datum/controller/process/mobs/parent)
+		if (..())
+			return TRUE
+
+		//Dogs bark sometimes
+		if (src.ai?.enabled && prob(5))
+			src.emote("scream", TRUE)
+
+	critter_attack(var/the_target)
+		if (istype(the_target, /mob))
+			var/mob/target = the_target
+			var/datum/targetable/critter/pounce/pounce = src.abilityHolder.getAbility(/datum/targetable/critter/pounce)
+			if (!pounce.disabled && pounce.cooldowncheck() && prob(50))
+				src.visible_message("<span class='combat'><B>[src]</B> barrels into [target] and trips them!</span>", "<span class='combat'>You run into [target]!</span>")
+				pounce.handleCast(target)
+				return
+			src.visible_message("<span class='combat'><B>[src]</B> bites [target]!</span>", "<span class='combat'>You bite [target]!</span>")
+			playsound(src.loc, 'sound/items/eatfoodshort.ogg', 50, 1, -1, 1.3)
+			random_brute_damage(target, src.attack_damage, 1)
+
+	disposing()
+		. = ..()
+		UnregisterSignal(src, COMSIG_MOB_THROW_ITEM_NEARBY)
+
+	proc/throw_response(target, obj/item/item, mob/thrower)
+		// Only ai dogs should play fetch
+		if (src == thrower || is_incapacitated(src) || !istype(item, /obj/item) || !src.ai?.enabled || prob(30) || length(src.ai.priority_tasks) > 0)
```

Comment:
```suggestion
		if (src == thrower || is_incapacitated(src) || !istype(item) || !src.ai?.enabled || prob(30) || length(src.ai.priority_tasks) > 0)
```
1-arg istype works here since the argument is typed.
Also, personal style, but I think it'd look clearer if the `prob(30)` was separated out into its own check. Odd to have all the sanity checks grouped with a random 30% chance to fail.